cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -162,11 +162,14 @@ public KafkaAssemblyOperator(Vertx vertx, boolean isOpenShift,
         }
         createReconciliationState(reconciliation, kafkaAssembly)
                 .reconcileCas()
+                // manual actions driven by the user come first
+                .compose(state -> state.zkManualPodCleaning())
+                .compose(state -> state.zkManualRollingUpdate())
+                .compose(state -> state.kafkaManualPodCleaning())
+                .compose(state -> state.kafkaManualRollingUpdate())
                 .compose(state -> state.clusterOperatorSecret())
```

Comment:
You shouldn't move the ZK rolling before the CO has generated a cluster CA-signed cert, otherwise it's possible that the ZK rolling can't find the ZK leader because it doesn't have a valid cert.

Reply:
Good point.

Reply:
I only know about this because it was something I ran into recently and had to fix.